website/docs/opengl/c1/2-hello-quad.md

```diff
@@ -0,0 +1,524 @@
+
+### Creating the buffer
+Now that we've got our vertex data, let's create the buffer!
+
+At the top of your class, add the following:
+
+```cs
+private static uint _vbo;
+```
+
+Then, in your `OnLoad` method, under where the vertices are defined, add:
+
+```
+_vbo = _gl.GenBuffer();
+_gl.BindBuffer(BufferTargetARB.ArrayBuffer, _vbo);
+```
+
+Much like the VAO, we need to first generate the buffer, and then bind it. Unlike vertex arrays, however, buffers need to be bound to a **target**. This allows you to bind buffers to different targets at the same time. Some of the more common targets include:
+
+* `ArrayBuffer` - The vertex buffer target (which is what we're using here).
+* `ElementArrayBuffer` - The element array buffer target (which you'll see later).
+* `UniformBuffer` - A uniform buffer, not used in this tutorial, but we'll get to it in a later tutorial.
+
+In this case, we're binding to `ArrayBuffer` since we're creating a vertex buffer.
+
+### Filling our buffer with data
+Let's fill our buffer with some data! Before we do that though, you need to be aware of `unsafe` in C#.
+
+#### Unsafe C#
+Silk.NET heavily uses `unsafe` code. Don't worry, this won't make your computer explode, however it does exit out of the "safe" managed environment of C#, and enters a realm where undefined behaviour, segmentation faults, and strange results are more likely to occur if you are not careful.
+
+<?# Info "If you wish to use `Span` instead, and remain in `safe` mode, Silk.NET does support these too. However, I will be using `unsafe` in this tutorial instead, as this is both what I personally use, as well as what the samples use." /?>
+
+Unsafe mode is not enabled by default, so we need to enable it. To enable it, add `<AllowUnsafeBlocks>true</AllowUnsafeBlocks>` to the `PropertyGroup` in your project's `.csproj` file. You also **MUST** add `unsafe` to your `OnLoad` and `OnRender` methods, like so:
+
+```cs
+public static unsafe void OnLoad() { ... }
+```
+
+If you forget to do either of these, you'll get compile errors. While our `OnRender` method does not contain any unsafe code yet, it will when we draw to the screen later, so it's best just to do it now so you don't forget to do it later.
+
+Now let's fill the buffer!
+
+Add the following after you create the buffer:
+
+```cs
+fixed (float* buf = vertices)
+    _gl.BufferData(BufferTargetARB.ArrayBuffer, (nuint) (vertices.Length * sizeof(float)), buf, BufferUsageARB.StaticDraw);
+```
+
+Let's go over what's going on here. First, we `fix` the vertices data. This prevents the garbage collector from moving it around, so we can take a pointer to the data. We tell it the target we want, the `ArrayBuffer` in this case, give it the data length, the buffer pointer, and we choose `StaticDraw` for our usage.
+
+Now, run the program again, and if you get the same blue window, you've set the buffer data successfully!
```

Comment:
(Not necessarily true, the blue window will remain whether you did it right or wrong)

Reply:
> (Note: Not necessarily true, the blue window will remain whether you did it right or wrong)

The program may also segfault/throw an exception (along with the driver doing who knows what when a GL usage error occurs)

Reply:
True that, but looking at the code it's waaaaay more likely that if you did something wrong it's gonna be something like forgetting to bind something, in which case it'll just silently fail.

I was just pointing it out though, I don't think it's wrong if that line remains as is

Reply:
The amount of times I have had driver crashes, segfaults, etc from failing to call `BufferData` correctly leads me to believe that if you do it wrong the program will crash.

You are right though that if the reader forgot to bind the buffer then they won't know, but that's why I include the "compare your code with the result" sections.

Reply:
Really? I don't remember ever being annoyed by segfaults or driver crashes from calling `BufferData`. My most common mistake is probably getting the size wrong, by forgetting to multiply by sizeof(whatever), which wouldn't cause any issues other than leaving part of the buffer with garbage. I do see why it could be a common problem tho